app/lib/search_facet_populator.rb

```diff
@@ -25,32 +27,43 @@ def self.populate_facet_by_name(facet_name, schema_object)
 
     updated_facet = SearchFacet.find_or_initialize_by(name: facet_name)
     updated_facet.identifier = facet_name
+    updated_facet.data_type = field_def['type'] == 'array' ? field_def['items']['type'] : field_def['type']
     updated_facet.is_ontology_based = is_ontology_based
     updated_facet.is_array_based = 'array'.casecmp(field_def['type']) == 0
-    updated_facet.is_numeric = 'number'.casecmp(field_def['type']) == 0
     updated_facet.big_query_id_column = facet_name
     updated_facet.big_query_name_column = is_ontology_based ? ontology_label_field_name : facet_name
     updated_facet.convention_name = schema_object['title']
     updated_facet.convention_version = alexandria_convention_config[:version]
-    # for now, just set the ontology name to the URL, since IIRC we're delegating display responsibility to the front end
-    updated_facet.ontology_urls = is_ontology_based ? [{name: field_def['ontology'], url: field_def['ontology']}] : []
+    if is_ontology_based
+      url = field_def['ontology']
+      ontology = fetch_json_from_url(url)
+      # check if response has expected keys; if not, default to URL for name value
+      ontology_name = ontology.dig('config', 'title') ? ontology['config']['title'] : url
```

Comment:
this is pretty slick!  If it's not too much of a hassle, you might as well add an assert for it around line 14 of search_facet_populator_test to check that we got the right ontology name for diseases

Reply:
Can do.

Reply:
fbc10ea